app/src/test/java/com/github/multimatum_team/multimatum/AddDeadlineTest.kt

```diff
@@ -0,0 +1,83 @@
+package com.github.multimatum_team.multimatum
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.contrib.PickerActions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.multimatum_team.multimatum.repository.DeadlineRepository
+import com.github.multimatum_team.multimatum.util.MockDeadlineRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import dagger.hilt.components.SingletonComponent
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.shadows.ShadowToast
+import javax.inject.Singleton
+
+
+/**
+ * Class test for AddDeadlineActivity
+ */
+@UninstallModules(RepositoryModule::class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4::class)
+class AddDeadlineTest {
+
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+
+    @get:Rule
+    val activityRule = ActivityScenarioRule(AddDeadlineActivity::class.java)
+
+    @Before
+    fun setUp() {
+        hiltRule.inject()
+    }
+
+    @Test
+    fun `The button should send a Toast if there is no title for the deadline`() {
+        Espresso.onView(ViewMatchers.withId(R.id.add_deadline_select_title))
+            .perform(ViewActions.replaceText(""))
+        Espresso.closeSoftKeyboard()
+        Espresso.onView(ViewMatchers.withId(R.id.add_deadline_button)).perform(ViewActions.click())
+        MatcherAssert.assertThat(
+            ShadowToast.getTextOfLatestToast(),
+            CoreMatchers.equalTo("Enter a title")
+        )
+    }
+
+    @Test
+    fun `The button should add a deadline given it's title and its date`() {
+        Espresso.onView(ViewMatchers.withId(R.id.add_deadline_select_title))
+            .perform(ViewActions.replaceText("Test 1"))
+        Espresso.closeSoftKeyboard()
+        Espresso.onView(ViewMatchers.withId(R.id.add_deadline_date_picker))
+            .perform(PickerActions.setDate(2022, 5, 8))
+        Espresso.onView(ViewMatchers.withId(R.id.add_deadline_button)).perform(ViewActions.click())
+        MatcherAssert.assertThat(
+            ShadowToast.getTextOfLatestToast(),
+            CoreMatchers.equalTo("Deadline created.")
```

Comment:
If you turn "Deadline created" into a resource as I suggested, you can replace it here too so that it is easier to change the message. The same holds for "Enter a title" in the previous test.

Reply:
Unfortunatly, you can't use the resource of string.xml in a test. It is not possible to use `getString()` unless you use a `@Mock`. I think however it is a good idea to keep it like that to check that the code correctly recuperate the resource.

Reply:
Does `R.string.name_of_the_string_resource` not work in tests?

Reply:
Oh OK I think I see the problem. You can use `applicationContext.getString(...)`

Reply:
Unfortunatly, `applicationContext.getString(...)` doesn't work either

Reply:
You should be able to use `RuntimeEnvironment.getApplication().applicationContext.getString(R.string.<your_resource>)` (hopefully it works this time...)

Reply:
This works ! Thanks !